drivers/ethernet/Kconfig.e1000

```diff
@@ -0,0 +1,28 @@
+# Kconfig - Intel(R) PRO/1000 Gigabit Ethernet driver configuration options
+
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig ETH_E1000
+	bool
+	prompt "Intel(R) PRO/1000 Gigabit Ethernet driver"
+	depends on NET_L2_ETHERNET PCI_ENUMERATION
```

Comment:
Is this correct? I don't remember ever seeing such syntax. However, it's easy to see following syntaxes:

~~~
timer/Kconfig:	depends on SOC_FAMILY_NRF && CLOCK_CONTROL_NRF5
~~~

and

~~~
Kconfig.nrfx:

config SPI_0_NRF_SPIS
        bool "nRF SPIS 0"
        depends on HAS_HW_NRF_SPIS0
        depends on SPI_SLAVE
~~~

I may imagine when options depends on 2 completely unrelated options, like here (Ethernet vs PCI), it's more readable to have 2 separated "depends on" lines, like the last example.

Calling @ulfalizer for suggestions.